src/praisonai-agents/tests/unit/test_gateway_hook_signature.py

```
@@ -0,0 +1,114 @@
+
+import pytest
+
+from praisonaiagents.gateway import HookConfig, verify_webhook_signature
+
+
+def _sign(secret: str, body: bytes, *, algo: str = "sha256", prefix: str = "") -> str:
+    return f"{prefix}{hmac.new(secret.encode(), body, algo).hexdigest()}"
+
+
+def test_verify_webhook_signature_valid_and_invalid():
+    secret = "s3cr3t"
```

Ruff (0.15.21)

[error] Possible hardcoded password assigned to: "secret"

(S105)